### ISSUE TITLE
Update "propdp" snippet to use nameof

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#44551]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1046667/update-propdp-snippet-to-use-nameof.html)._

---
The `propdp` C# code snippet creates the following code:

```
public int MyProperty
{
    get { return (int)GetValue(MyPropertyProperty); }
    set { SetValue(MyPropertyProperty, value); }
}

// Using a DependencyProperty as the backing store for MyProperty.  This enables animation, styling, binding, etc...
public static readonly DependencyProperty MyPropertyProperty =
    DependencyProperty.Register("MyProperty", typeof(int), typeof(ownerclass), new PropertyMetadata(0));
```

Notice the `"MyProperty"` string literal, which points to the backing property for the dependency property. This is not ideal, because renaming the property will not change this string and will cause unexpected problems at runtime. Instead, the code snippet could generate the following:

```
...
public static readonly DependencyProperty MyPropertyProperty =
    DependencyProperty.Register(nameof(MyProperty), typeof(int), typeof(ownerclass), new PropertyMetadata(0));
```

Using the `nameof` operator, the code would gain compile time safety and would automatically update the reference whenever the property name changes.

---
#### Visual Studio Feedback System on 5/25/2020, 03:06 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by Youssef1313** [comment]
Where does the snippets exist in the repo?
I can't find them.

**Comment by DaveInCaz** [comment]
I'm curious why this got filed under the roslyn project? Isn't' that the compiler engine? This just seems like a Visual Studio feature.

**Comment by CyrusNajmabadi** [comment]
Roslyn supplies the VS IDE side of thing for C#/VB and portions of F#/TS/JS/Xaml.

**Comment by sharwell** [comment]
Blocked on #31879

**Comment by DaveInCaz** [comment]
FWIW there is a StackOverflow post which gets some notice about this same issue. It proposes a local workaround for developers to modify the snippet file. 

https://stackoverflow.com/q/35392084/3195477

**Comment by DaveInCaz** [comment]
Is this a dupe of https://github.com/dotnet/roslyn/issues/5574 ?

**Comment by sharwell** [comment]
Duplicate of #5574
